ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/ImageNameTd.tsx

```diff
@@ -17,13 +19,52 @@ export type ImageNameTdProps = {
 
 function ImageNameTd({ name, id, children }: ImageNameTdProps) {
     const vulnerabilityState = useVulnerabilityState();
+    const [copyIconTooltip, setCopyIconTooltip] = useState('Copy image name');
+    const [resetTooltip, cancelTimeout] = useTimeout(() => setCopyIconTooltip('Copy image name'));
+
+    useEffect(() => cancelTimeout, [cancelTimeout]);
+
+    const { registry, remote, tag } = name;
+
+    function copyImageName() {
+        navigator?.clipboard
+            ?.writeText(`${registry}/${remote}:${tag}`)
+            .then(() => setCopyIconTooltip('Copied!'))
+            .catch(() => {}) /* Nothing to do */
+            .finally(() => resetTooltip(2000));
+    }
+
     return (
-        <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsNone' }}>
-            <Link to={getEntityPagePath('Image', id, vulnerabilityState)}>
-                <Truncate position="middle" content={`${name.remote}:${name.tag}`} />
-            </Link>{' '}
-            <span className="pf-u-color-200 pf-u-font-size-sm">in {name.registry}</span>
-            <div>{children}</div>
+        <Flex
+            direction={{ default: 'row' }}
+            flexWrap={{ default: 'nowrap' }}
+            justifyContent={{ default: 'justifyContentSpaceBetween' }}
+            alignItems={{ default: 'alignItemsFlexStart' }}
+            spaceItems={{ default: 'spaceItemsNone' }}
+        >
+            <Flex
+                grow={{ default: 'grow' }}
+                direction={{ default: 'column' }}
+                spaceItems={{ default: 'spaceItemsNone' }}
+            >
+                <Link to={getEntityPagePath('Image', id, vulnerabilityState)}>
+                    <Truncate position="middle" content={`${remote}:${tag}`} />
+                </Link>{' '}
+                <span className="pf-u-color-200 pf-u-font-size-sm">in {registry}</span>
+                <div>{children}</div>
+            </Flex>
+            <FlexItem shrink={{ default: 'shrink' }}>
+                <Tooltip content={copyIconTooltip} position="top">
+                    <Button
+                        isSmall
+                        className="pf-u-pt-xs"
+                        variant="plain"
+                        aria-label="Copy image name to clipboard"
+                        icon={<CopyIcon />}
+                        onClick={copyImageName}
+                    />
+                </Tooltip>
```

Comment:
Observing, not necessarily requesting change, that a PatternFly element might encapsulate the tooltip.
https://www.patternfly.org/components/clipboard-copy#clipboardcopybutton

The timeout did not occur to me in this occurrence:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Components/PatternFly/ErrorBoundary/ErrorBoundaryCodeBlock.tsx#L41-L49

Reply:
Aha! I looked at `ClipboardCopy` but disregarded it due to the enforced styles. I didn't realize we could get just the button for free.

I'll try that, and if it works it might remove the need for the timeout as well.

Reply:
For error boundary, I admit that I was very conservative about any TypeScript code on the page.